plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -165,98 +201,187 @@ def main():
+        'reporter': gitlab.REPORTER_ACCESS,
+        'developer': gitlab.DEVELOPER_ACCESS,
+        'maintainer': gitlab.MAINTAINER_ACCESS,
+        'owner': gitlab.OWNER_ACCESS
+    }
+
     gitlab_group = module.params['gitlab_group']
-    gitlab_user = module.params['gitlab_user']
     state = module.params['state']
     access_level = module.params['access_level']
+    purge_users = module.params['purge_users']
 
     # convert access level string input to int
     if access_level:
-        access_level_int = {
-            'guest': gitlab.GUEST_ACCESS,
-            'reporter': gitlab.REPORTER_ACCESS,
-            'developer': gitlab.DEVELOPER_ACCESS,
-            'maintainer': gitlab.MAINTAINER_ACCESS,
-            'owner': gitlab.OWNER_ACCESS
-        }
-
         access_level = access_level_int[access_level]
 
+    if purge_users:
+        for i in range(len(purge_users)):
+            purge_users[i] = access_level_int[purge_users[i]]
+
     # connect to gitlab server
     gl = gitlabAuthentication(module)
 
     group = GitLabGroup(module, gl)
 
-    gitlab_user_id = group.get_user_id(gitlab_user)
     gitlab_group_id = group.get_group_id(gitlab_group)
 
     # group doesn't exist
     if not gitlab_group_id:
         module.fail_json(msg="group '%s' not found." % gitlab_group)
 
-    # user doesn't exist
-    if not gitlab_user_id:
-        if state == 'absent':
-            module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
-        else:
-            module.fail_json(msg="user '%s' not found." % gitlab_user)
-
-    members = group.get_members_in_a_group(gitlab_group_id)
-    is_user_a_member = group.is_user_a_member(members, gitlab_user_id)
-
-    # check if the user is a member in the group
-    if not is_user_a_member:
-        if state == 'present':
-            # add user to the group
-            if not module.check_mode:
-                group.add_member_to_group(gitlab_user_id, gitlab_group_id, access_level)
-            module.exit_json(changed=True, result="Successfully added user '%s' to the group." % gitlab_user)
-        # state as absent
-        else:
-            module.exit_json(changed=False, result="User, '%s', is not a member in the group. No change to report" % gitlab_user)
-    # in case that a user is a member
+    members = []
+    gitlab_users_access = []
+    if module.params['gitlab_user'] is not None:
+        gitlab_users = module.params['gitlab_user']
+        for glu in gitlab_users:
+            gitlab_users_access.append({'name': glu, 'access_level': access_level})
+    elif module.params['gitlab_users_access'] is not None:
+        gitlab_users_access = module.params['gitlab_users_access']
+        for i in range(len(gitlab_users_access)):
+            if gitlab_users_access[i]['access_level']:
+                gitlab_users_access[i]['access_level'] = access_level_int[gitlab_users_access[i]['access_level']]
+
```

Comment:
You need to do some validation that the same user isn't mentioned multiple times.

Reply:
What problem do you see if a user is mentioned multiple times? Worst case I see is he will be assigned to access level A first and to B afterwards.

Reply:
It will cause constant changes for that user, and users will have potentially a long time until they figure out why that happens. (And probably create an issue here and someone from community has to figure it out for them.)

Reply:
Only mapping access level will not do any change to the user.